### DISCUSSION TITLE
SonoffSV with BME280 and reed switch controlling garage door

### DISCUSSION THREAD

**Discussion by deadrabbit87** [arendst/Tasmota#17875] [Q&A]
Hi,

for controlling my garage door, i want to use an Sonoff SV. The internal relay is switching the garage motor. Therefore i configured `Power1` with `PulseTime 10` to give the garage motor a trigger. This is working as expected. 

Also i've connected an BME280 to `GPIO 4` and `GPIO 5`. This works also fine and Tasmota is publishing the temperature via mqqt to HomeAssistant. 

What i'm struggling with, is the reed switch: For displaying the current state of the garage door, i've wired a reed switch to the leftover `GPIO 14`. As far is i know, i have to configure `GPIO 14` as another switch (`Switch2`) and a virtual relay (`relay2`) on `GPIO 3`: 

![grafik](https://user-images.githubusercontent.com/24924801/216759289-4ef60510-8e75-4121-b698-a8b045ce604f.png)

Setting up `relay2` gives me `Power2` which i'm able to switch. But this relay is not connected to `Switch2`. As i understand, this should be the case? 

What i want in the end: A simple sensor in HomeAssistant displaying the state of my garage door. 

BW Maximilian

**Comment by barbudor** [comment]
What is your switch mode for switch 2?
Could you set Weblog 3 and provide logs when actuating the Reed switch?

**Reply by barbudor** [reply]
While I don't have any knowledge of HA, I've heard that the best is to use only the switch with switchmode2 15 (to detach) and create a 'binary sensor' in HA using Yaml description. Don't ask how to do that Yaml description, I have no idea 😂

**Reply by deadrabbit87** [reply]
Yes, of course: 

```
11:14:18.130 SRC: Rule
11:14:18.133 CMD: Grp 0, Cmd 'PUBLISH', Idx 1, Len 41, Pld -99, Data 'stat/SonoffSVGarage/RESULT {"POWER2":"0"}'
11:14:18.138 MQT: stat/SonoffSVGarage/RESULT = {"POWER2":"0"}
11:14:24.098 RUL: SWITCH2#STATE performs "publish stat/SonoffSVGarage/RESULT {"POWER2":"1"}"
11:14:24.100 SRC: Rule
11:14:24.103 CMD: Grp 0, Cmd 'PUBLISH', Idx 1, Len 41, Pld -99, Data 'stat/SonoffSVGarage/RESULT {"POWER2":"1"}'
11:14:24.108 MQT: stat/SonoffSVGarage/RESULT = {"POWER2":"1"}
11:15:30.434 CMD: switchmode2
11:15:30.436 SRC: WebConsole from XXXXXXXXXXXX
11:15:30.438 CMD: Grp 0, Cmd 'SWITCHMODE', Idx 2, Len 0, Pld -99, Data ''
11:15:30.443 MQT: stat/SonoffSVGarage/RESULT = {"SwitchMode2":1}
```

What i've forgot to say is, that i created a rule for publishing the state of `Power2` via mqqt according to [this Wiki article](https://tasmota.github.io/docs/Rules/#send-mqtt-message-on-button-press).

**Reply by barbudor** [reply]
What are this rules and why?
It looks like you are publishing power 2 commands to yourself which is useless
Using rules on switch 2 detach the switch from the relay then you are using rules to send power commands to the same relay
That sounds over engineered for something that is supposed to work from scratch

**Reply by deadrabbit87** [reply]
Using the rules was only clutching at straws to get this to work. 

What i've done now: Making a entire reset of the Sonoff SV and configured it from bottom up. 

I didn't know why, but now it is working...  It is exactly the same configuration as i wrote above, but now it is working: 
![grafik](https://user-images.githubusercontent.com/24924801/216765210-26c368b8-bb81-4d69-b9a8-8e4fb5f87a66.png)
`Power2` is switching from on to off if i trigger the reed switch. 

Now i'm able to integrate this into HA. 

Thanks!

**Reply by barbudor** [reply]
Great to know it's working now

When experimenting, you may change config in a way you don't remember where you stand.
Resetting and Restarting from scratch can generally help to clear once mind and doubts

**Comment by Noschvie** [comment]
My question regarding reed switch configuration : is there a paramter for the debounce time ? Maybe 2 seconds. Thanks!

**Reply by barbudor** [reply]
In the above setup, he is using a Switch.
So the command is [SwitchDebounce](https://tasmota.github.io/docs/Commands/#switchdebounce)
